INSTALL.txt

```diff
@@ -1,30 +1,29 @@
   Build requirements (Installing from source)
-    For building Rakudo you need at least a C compiler, a "make" utility,
-    and Perl 5.10 or newer. To automatically obtain and build MoarVM
-    as well as NQP, you may also need a git client, which is also needed
-    for fetching the test suite.
+    For building Rakudo you need at least a C compiler, a "make" utility, and
+    Perl 5.10 or newer. To automatically obtain and build MoarVM as well as NQP,
```

Comment:
On Windows we now require >= Perl 5.22 ([370956](https://github.com/rakudo/rakudo/commit/370956b913ce5eac1d89a17626ebbdf99cc0170a)).

Reply:
Did you see [my comment](https://github.com/Raku/nqp-configure/issues/19#issuecomment-627672367)? With simplified nmake detection this could now be less of a problem, but the build sample provided by @Altai-man doesn't contain any nmake output at all! If anybody can confirm that 5.18 doesn't have the problem I'd consider sticking to that version for a while.

Besides, I don't see a reason why older versions of Perl won't work for Windows.

What certainly makes sense is changing to 5.10.1 as this is what stated in Config.pm.

Reply:
The reason the dependency was bumped to 5.22 for Windows is the use of the [pipe form of open](https://github.com/rakudo/rakudo/blob/master/tools/build/update-submodules.pl#L71) which is only supported in Perl 5.22 and later on Windows. I think that's a reason unrelated to the one you had in mind.

Reply:
I'm not really happy about bumping to 5.22 only for the sake of a single `open` call. Can we get around it?

Reply:
Bumping to 5.22 for that sounds harsh, but I also wonder if this would ever cause a problem. 5.22 is 5 years old already. On Windows there's no such thing as "system perl" that is stuck on some ancient version and still gets support for another 5 years. Every perl on Windows has been installed manually and will probably have been the most recent version at that time. So everything **installed** during the last 5 years will be OK.

Reply:
The problem is that the script is used by every build, not only Windows. But as far as it is Windows only requirement a specific note can be added.

@patrickbkr what would you say?

Reply:
It is a Windows only requirement. `open` with pipe is a feature that's been in Perl for a long time. Only on Windows the support was added in 5.22.

There was quite a lot of discussion and exploration of different solutions for this very line of code, all having their tradeoffs (if there is interest I can dig up the discussion). In the end we settled on this solution with the tradeoff of having to bump the Perl version on Windows. So from what I remember there is no much nicer solution.

Reply:
Ok, updated with Perl 5.22 for Windows.

Heh, I was missing named capture call in Perl regexes while implementing macros...